Feature/chefs

keren-elbaz-m opened the pull request.

keren-elbaz-m committed:
Chefs

Uploaded images for chefs

keren-elbaz-m committed:
API route

Created API routes to get all chefs and get chef by ID

keren-elbaz-m committed:
fixed the image name

keren-elbaz-m committed:
Create getRandomChef.ts

create helper for random chef of the week and apply responsive layout

keren-elbaz-m committed:
Update text.ts

add RESOURCES strings for chef of the week and chef restaurants title

keren-elbaz-m committed:
Update index.ts

add CHEF_RESTAURANT enum to SectionPart

keren-elbaz-m committed:
implement "Chef of the Week" feature with dynamic API and responsive layout

Created API routes to fetch chef by ID and their restaurants
Implemented getItemFromApi helper
Added SectionPart.CHEF_RESTAURANT enum for new section type
Updated MapToCards to support chef description
Adjusted layout and data flow in page.tsx

keren-elbaz-m committed:
update responsive layout and styles for Chef section

Adjusted Card and Section components for Chef layout
Made .cardWrapper--chef responsive in order to use 2-column grid

keren-elbaz-m committed:
Fixed chefs names

And removed console.log from page.tsx

keren-elbaz-m committed:
Fixed chefs card

fixed the title and the chefs names

keren-elbaz-m committed:
extract helper function to get chef's first name

DanielBaronMoveo commented on src/app/globals.css:
```diff
@@ -25,7 +25,7 @@ body {
 
 * {
-  box-sizing: border-box;
+  /* box-sizing: border-box; */
```
We don't push commented out code to PRs.
Also, why did you make the following change?

keren-elbaz-m replied:
I removed box-sizing: border-box earlier thinking it was breaking the grid layout  for the chef card, but after testing, everything looks fine with it now, so I’m keeping it in.

DanielBaronMoveo replied:
Mainly, we want to keep it as initialized css property due to the reason we want to keep all the elements boxed.

DanielBaronMoveo commented on src/app/api/chef/[id]/restaurants/route.ts:
Why don't we "aggregate" the chefRestaurants within the chef object and just return it as a coupled object?
I assume right now we are making TWO API calls just to get the chef and it's restaurants.

keren-elbaz-m replied:
you right, I refactored the api call.

DanielBaronMoveo commented on src/components/Card/Card.tsx:
```diff
@@ -10,25 +10,40 @@ type CardProps = {
+
+                {variant !== SectionPart.CHEF && (
+                    <div className={`${styles.content} ${styles[`content--${variant}`]}`}>
+                        <h1 className={`${styles.name} ${styles[`name--${variant}`]}`}>{item.name}</h1>
```
Change ```<h1> to <h2>``` (cards typically shouldn't use h1 as they're not page titles)

keren-elbaz-m replied:
I see, changed it :)

DanielBaronMoveo commented on src/components/Card/Card.tsx:
```diff
@@ -10,25 +10,40 @@ type CardProps = {
+                )}
+
+                {variant === SectionPart.CHEF && (
+                    <h1 className={`${styles.name} ${styles[`name--${variant}`]}`}>{item.name}</h1>
```
Change ```<h1> to <h2>``` (cards typically shouldn't use h1 as they're not page titles)

keren-elbaz-m replied:
done

DanielBaronMoveo commented on src/components/Card/Card.tsx:
This whole component could be refactored to match better flexibility and be more robust.

Take some references from the following: 

```
type RestaurantItem = Restaurant;
type DishItem = {
  name: string;
  imageUrl: string;
  price?: number;
  description?: string;
};
type ChefItem = {
  name: string;
  imageUrl: string;
  specialty?: string;
  experience?: number;
};


type CardItem = RestaurantItem | DishItem | ChefItem;


type CardProps = {
  children?: React.ReactNode;
} & (
  | { variant: SectionPart.RESTAURANT; item: RestaurantItem }
  | { variant: SectionPart.DISH; item: DishItem }
  | { variant: SectionPart.CHEF | SectionPart.CHEF_RESTAURANT; item: ChefItem }
);

const getClassName = (
  base: string,
  variant?: SectionPart,
  additionalClasses?: string[]
): string => {
  const classes = [styles[base]];
  
  if (variant) {
    classes.push(styles[`${base}--${variant}`]);
  }
  
  if (additionalClasses) {
    classes.push(...additionalClasses);
  }
  
  return classes.filter(Boolean).join(' ');
};

export default function Card({ item, children, variant }: CardProps) {
  const isChefVariant = variant === SectionPart.CHEF;
  
  return (
    <article 
      className={getClassName('cardWrapper', variant)}
      aria-label={`${variant} card for ${item.name}`}
    >
      <div className={getClassName('card', variant)}>
        <div className={getClassName('imageContainer', variant)}>
          <Image
            src={item.imageUrl}
            alt={`${variant === SectionPart.CHEF ? 'Portrait of' : 'Image of'} ${item.name}`}
            fill
            sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
            priority={false}
            loading="lazy"
          />
        </div>

        {!isChefVariant ? (
          <div className={getClassName('content', variant)}>
            <h2 className={getClassName('name', variant)}>{item.name}</h2>
            {children && (
              <div className={getClassName('children', variant)}>
                {children}
              </div>
            )}
          </div>
        ) : (
          <h2 className={getClassName('name', variant)}>{item.name}</h2>
        )}
      </div>

      {isChefVariant && children && (
        <div className={getClassName('children', variant)}>
          {children}
        </div>
      )}
    </article>
  );
}

keren-elbaz-m replied:
Refactored the component :)

DanielBaronMoveo commented on src/constans/Api.constans.ts:
```diff
@@ -3,4 +3,7 @@ export const API_ROUTES = {
\ No newline at end of file
+}
+
+export const API_CHEF_BY_ID = (id: number | string) => `/chef/${id}`;
+export const API_REST_CHEF_BY_ID = (id: number | string) => `/chef/${id}/restaurants`;
```
I still think both of them should be combined with the getChefData.

keren-elbaz-m replied:
done :)

keren-elbaz-m committed:
Update globals.css

Restored global box-sizing rule after layout confirmed stable

keren-elbaz-m committed:
Refactored chef API to return full chef object including restaurants

Combined chef and restaurant data into a single API response from /api/chef/[id]
Replaced separate client-side calls to /chef/:id and /chef/:id/restaurants with a single call
Created 'services' folder under lib/ and moved chef and restaurant data access logic into:
chefService.ts and restaurantService.ts
Refactored lib folder: separated into 'services' and 'utils' directories

keren-elbaz-m committed:
Chaged the file name

Change the file name from getCardsFromApi.ts to getDataFromApi.ts

keren-elbaz-m committed:
Changed card title tag from <h1> to <h2>

keren-elbaz-m committed:
Refactored Card component

Converted Card component to use discriminated unions with precise types for RestaurantItem, DishItem, and ChefItem.
Added getClassName helper to dynamically construct style class names
Wrapped Card in <article> to semantically identify each card’s purpose.

keren-elbaz-m committed:
Refactored MapToCards

Added item shape validation
Used switch-case to cast items per SectionPart and return correct Card type
